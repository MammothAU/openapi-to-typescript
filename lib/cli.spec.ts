import execa from 'execa'

describe('openapi-to-typescript CLI', () => {
  it('should --help', async () => {
    const executed = await runCLI('--help')
    expect(executed.stdout).toMatchSnapshot()
  })
  it('should output typescipt code to stdout', async () => {
    const executed = await runCLI('-i ./fixtures/petstore.json')
    expect(executed.stdout).toContain('API_URL = ')
    expect(executed.stdout).toContain('ShowPetById')
  })
})

function runCLI(arg: string) {
  return execa('yarn', ['cli', ...arg.split(/\s+/g)])
}
