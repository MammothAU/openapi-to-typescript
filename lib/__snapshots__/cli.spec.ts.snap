// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`openapi-to-typescript CLI should --help 1`] = `
"[2K[1G[2m$ ts-node ./lib/cli --help[22m
cli

Write generated output to a file

Commands:
  cli generate  Write generated output to a file                       [default]

Options:
  --help         Show help                                             [boolean]
  --version      Show version number                                   [boolean]
  --input, -i    File to read from                                    [required]
  --output, -o   Output name to write to
  --verbose, -v                                                 [default: false]"
`;
